programs/buck_version.py

```diff
@@ -80,17 +80,13 @@ def get_clean_buck_version(dirpath, allow_dirty=False):
 
 
 def get_dirty_buck_version(dirpath):
-    git_tree_in = check_output(
-        ['git', 'log', '-n1', '--pretty=format:%T', 'HEAD', '--'],
-        cwd=dirpath).strip()
-
     with EmptyTempFile(prefix='buck-git-index') as index_file:
+        subprocess.check_call(
+            ['git', 'read-tree', '--index-output', index_file.name, '--reset', 'HEAD'],
```

Comment:
I glossed right over the part about writing over the file.  That's what I get for looking at changes right when I get in :)

This is fine as-is.